File: src/mcp/tool_servers/python_tool_server/health.py
"""Tool server health monitoring."""

from typing import Dict, List

from mcp_core import HealthCheck, HealthStatus, logger


class ToolHealth(HealthCheck):
    """Tool server health check implementation."""

    def __init__(self):
        self.tools = {}

    async def check_health(self) -> Dict:
        """Check tool server health."""
        try:
            # Check tool availability
            tools = await self.check_tools()

            return {
                "status": HealthStatus.HEALTHY,
                "details": {"active_tools": len(tools), "tools": tools},
            }
        except Exception as e:
            logger.error("Tool health check failed", error=str(e))
            return {"status": HealthStatus.UNHEALTHY, "error": str(e)}

    async def check_tools(self) -> List[str]:
        """Check available tools."""
        return list(self.tools.keys())

    def register_tool(self, name: str):
        """Register a tool for health monitoring."""
        self.tools[name] = {"status": "registered", "last_check": None}

File: src/mcp/tool_servers/typescript_tool_server/src/health.ts
import { HealthCheck } from '@modelcontextprotocol/sdk/types';
import { logger } from './logger';

export const healthCheck: HealthCheck = {
  async check() {
    try {
      // Check critical services
      await checkTools();

      return {
        status: 'healthy',
        details: {
          tools: 'available',
        },
      };
    } catch (error) {
      logger.error('Health check failed', { error });

      return {
        status: 'unhealthy',
        error: String(error),
      };
    }
  },
};

async function checkTools(): Promise<void> {
  // Implement tool availability check
  return Promise.resolve();
}
